ci-operator/step-registry/openshift/microshift/e2e/metal-tests/openshift-microshift-e2e-metal-tests-commands.sh

```diff
@@ -3,6 +3,59 @@ set -xeuo pipefail
 
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
 
+finalize() {
+  scp -r "${INSTANCE_PREFIX}:/home/${HOST_USER}/microshift/_output/test-images/scenario-info" "${ARTIFACT_DIR}"
+
+  STEP_NAME="${HOSTNAME##${JOB_NAME_SAFE}-}"
+  REPORT="${ARTIFACT_DIR}/custom-link-tools.html"
+  cat >>${REPORT} <<EOF
+<html>
+<head>
+  <title>Test logs</title>
+  <meta name="description" content="Links to relevant logs">
+  <link rel="stylesheet" type="text/css" href="/static/style.css">
+  <link rel="stylesheet" type="text/css" href="/static/extensions/style.css">
+  <link href="https://fonts.googleapis.com/css?family=Roboto:400,700" rel="stylesheet">
+  <link rel="stylesheet" href="https://code.getmdl.io/1.3.0/material.indigo-pink.min.css">
+  <link rel="stylesheet" type="text/css" href="/static/spyglass/spyglass.css">
+  <style>
+    a {
+        display: inline;
+        color: #FFFFFF;
+    }
+    a:hover {
+      text-decoration: underline;
+    }
+    p {
+      display: inline;
+      color: #FFFFFF;
+    }
+  </style>
+</head>
+<body>
+EOF
+  for test in ${ARTIFACT_DIR}/scenario-info/*; do
+    testname=$(basename "${test}")
+    cat >>${REPORT} <<EOF
+    <p>${testname}</p>:&nbsp;
+    <a target="_blank" href="https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/${REPO_OWNER}_${REPO_NAME}/${PULL_NUMBER}/${JOB_NAME}/${BUILD_ID}/artifacts/${JOB_NAME_SAFE}/${STEP_NAME}/${ARTIFACT_DIR#/logs/}/scenario-info/${testname}">directory</a>
+    &nbsp;/&nbsp;<a target="_blank" href="https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/${REPO_OWNER}_${REPO_NAME}/${PULL_NUMBER}/${JOB_NAME}/${BUILD_ID}/artifacts/${JOB_NAME_SAFE}/${STEP_NAME}/${ARTIFACT_DIR#/logs/}/scenario-info/${testname}/run.log">run.log</a>
+EOF
+    if [ -f ${test}/log.html ]; then
+      cat >>${REPORT} <<EOF
+    &nbsp;/&nbsp;<a target="_blank" href="https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/${REPO_OWNER}_${REPO_NAME}/${PULL_NUMBER}/${JOB_NAME}/${BUILD_ID}/artifacts/${JOB_NAME_SAFE}/${STEP_NAME}/${ARTIFACT_DIR#/logs/}/scenario-info/${testname}/log.html">RF log</a>
+EOF
+    fi
+    cat >>${REPORT} <<EOF
+    <br>
+EOF
```

Comment:
It's not a big deal and diff doesn't show it correctly, but small suggestion - honestly, if you don't like it, you can ignore it :smile: 
- remove `display: inline` from `p` styling
- remove `</pr>` after `${testname}`
- change `<br>` to `</pr>`

```suggestion
    p {
      color: #FFFFFF;
    }
  </style>
</head>
<body>
EOF
  for test in ${ARTIFACT_DIR}/scenario-info/*; do
    testname=$(basename "${test}")
    cat >>${REPORT} <<EOF
    <p>${testname}:&nbsp;
    <a target="_blank" href="https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/${REPO_OWNER}_${REPO_NAME}/${PULL_NUMBER}/${JOB_NAME}/${BUILD_ID}/artifacts/${JOB_NAME_SAFE}/${STEP_NAME}/${ARTIFACT_DIR#/logs/}/scenario-info/${testname}">directory</a>
    &nbsp;/&nbsp;<a target="_blank" href="https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/${REPO_OWNER}_${REPO_NAME}/${PULL_NUMBER}/${JOB_NAME}/${BUILD_ID}/artifacts/${JOB_NAME_SAFE}/${STEP_NAME}/${ARTIFACT_DIR#/logs/}/scenario-info/${testname}/run.log">run.log</a>
EOF
    if [ -f ${test}/log.html ]; then
      cat >>${REPORT} <<EOF
    &nbsp;/&nbsp;<a target="_blank" href="https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/${REPO_OWNER}_${REPO_NAME}/${PULL_NUMBER}/${JOB_NAME}/${BUILD_ID}/artifacts/${JOB_NAME_SAFE}/${STEP_NAME}/${ARTIFACT_DIR#/logs/}/scenario-info/${testname}/log.html">RF log</a>
EOF
    fi
    cat >>${REPORT} <<EOF
    </p>
EOF
```

Reply:
Done.